tests/utils/checks.py

```diff
@@ -66,3 +73,52 @@ def assert_is_identical_hetero(g, g2, ignore_internal_data=False):
         assert len(g.edges[etype].data) == len(g2.edges[etype].data)
         for k in g.edges[etype].data:
             assert F.allclose(g.edges[etype].data[k], g2.edges[etype].data[k])
+
+
+def check_graph_equal(g1, g2, *, check_idtype=True, check_feature=True):
+    assert g1.device == g1.device
+    if check_idtype:
+        assert g1.idtype == g2.idtype
+    assert g1.ntypes == g2.ntypes
+    assert g1.etypes == g2.etypes
+    assert g1.srctypes == g2.srctypes
+    assert g1.dsttypes == g2.dsttypes
+    assert g1.canonical_etypes == g2.canonical_etypes
+    assert g1.batch_size == g2.batch_size
+
+    # check if two metagraphs are identical
+    for edges, features in g1.metagraph().edges(keys=True).items():
+        assert g2.metagraph().edges(keys=True)[edges] == features
+
+    for nty in g1.ntypes:
+        assert g1.number_of_nodes(nty) == g2.number_of_nodes(nty)
```

Comment:
```suggestion
        assert g1.num_nodes(nty) == g2.num_nodes(nty)
```

Reply:
Not related to this PR, will have a separate PR to fix them all.